File: aml.client/src/app/_services/branch.service.ts
import { HttpClient, HttpHeaders, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Branch } from "../_models/Branch";
import { Observable } from "rxjs";
import { UpdateBranchRequest } from "../_DTOs/UpdateBranchRequest";

const AUTH_API = 'http://localhost:7033/api/branch/';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable({
  providedIn: 'root',
})

export class BranchService {
  constructor(private http: HttpClient) { }

  getBranches(): Observable<any> {
    return this.http.get(
      AUTH_API + 'get-branches',
      httpOptions
    );
  }

  getBranch(branchId: any): Observable<any> {
    let params = new HttpParams().set("branchId", branchId)
    let httpOptions = {
      headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
      params
    };

    return this.http.get(
      AUTH_API + 'get-branch',
      httpOptions
    );
  }

  updateBranch(email: string, branch: Branch): Observable<any> {
    const request: UpdateBranchRequest = {
      email: email,
      branch: branch
    }

    return this.http.post(
      AUTH_API + 'update-branch',
      request,
      httpOptions
    );
  }
}

File: AML.Server/Services/BranchService.cs
using AML.Server.DTOs;
using AML.Server.Interfaces;
using AML.Server.Models;

namespace AML.Server.Services
{
    public class BranchService : IBranchService
    {
        private readonly IBranchRepository _branchRepository;

        public BranchService(IBranchRepository branchRepository)
        {
            _branchRepository = branchRepository;
        }

        public async Task<List<Branch>> GetBranches()
        {
            return await _branchRepository.GetBranches();
        }

        public async Task<Branch> GetBranch(int branchId)
        {
            return await _branchRepository.GetBranch(branchId);
        }

        public async Task<Account> UpdateBranch(UpdateBranchRequest request)
        {
            return await _branchRepository.UpdateBranch(request.Email, request.Branch);
        }
    }
}
